### PR TITLE
Model by Path(URI) and with Chunking

### PR DESCRIPTION
**Pull request by sazzad16**
Different approach #39

### CODE REVIEW

**Comment by gkorland** [comment]
@sazzad16 I thin the default should be 500MB to make sure users won't get an error by default

**Comment by sazzad16** [comment]
@gkorland @DvirDukhan different approach #39

File: src/main/java/com/redislabs/redisai/Model.java [hunk 1]
```diff
@@ -1,24 +1,47 @@
...
 import redis.clients.jedis.util.SafeEncoder;
 
 /** Direct mapping to RedisAI Model */
 public class Model {
 
-  private Backend backend;
-  private Device device;
+  public static final String BLOB_CHUNK_SIZE_PROPERTY = "redisai.blob.chunkSize";
+
+  private static final int BLOB_CHUNK_SIZE =
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisai/Model.java:19]: @sazzad16 the default should not be "0" but 500MB

**Review by gkorland — CHANGES_REQUESTED** [review]

File: src/main/java/com/redislabs/redisai/Model.java [hunk 2]
```diff
@@ -314,11 +352,26 @@ public Model setMinBatchTimeout(long minBatchTimeout) {
...
 
     args.add(Keyword.BLOB.getRaw());
-    args.add(blob);
+    collectChunks(args, blob);
 
     return args;
   }
 
+  private static void collectChunks(List<byte[]> collector, byte[] array) {
+    final int chunkSize = BLOB_CHUNK_SIZE;
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisai/Model.java:361]: @sazzad16 why do you need this local variable and not use the constant?

File: src/main/java/com/redislabs/redisai/Model.java [hunk 3]
```diff
@@ -1,24 +1,47 @@
...
 
 /** Direct mapping to RedisAI Model */
 public class Model {
 
-  private Backend backend;
-  private Device device;
+  public static final String BLOB_CHUNK_SIZE_PROPERTY = "redisai.blob.chunkSize";
+
+  private static final int BLOB_CHUNK_SIZE =
+      Integer.parseInt(System.getProperty(BLOB_CHUNK_SIZE_PROPERTY, "0"));
```
Comment (sazzad16) [inline src/main/java/com/redislabs/redisai/Model.java:20]:
```suggestion
      Integer.parseInt(System.getProperty(BLOB_CHUNK_SIZE_PROPERTY, "536870912"));
```

File: src/test/java/com/redislabs/redisai/ChunkTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  @Test
+  public void commandWithChunking() throws IOException, URISyntaxException {
+    Model model =
+        new Model(
+            Backend.ONNX,
+            Device.CPU,
+            getClass().getClassLoader().getResource("test_data/mnist.onnx").toURI());
+
+    try (RedisAI ai = new RedisAI()) {
```
Comment (gkorland) [inline src/test/java/com/redislabs/redisai/ChunkTest.java:50]: @sazzad16 can you please verify that the loaded model is still functional after it was loaded?

File: src/main/java/com/redislabs/redisai/Model.java (hunk 2 above)
Comment (sazzad16) [reply src/main/java/com/redislabs/redisai/Model.java:361]: @gkorland just for code readability. This variable is used several times and a smaller variable name seemed better.

**Review by gkorland — CHANGES_REQUESTED** [review]

File: src/main/java/com/redislabs/redisai/Model.java (hunk 2 above)
Comment (gkorland) [reply src/main/java/com/redislabs/redisai/Model.java:361]: I find it more confusing...

**Review by gkorland — APPROVED** [review]

**Review by gkorland — APPROVED** [review]

**Pull request merged**